### PR TITLE
Fix/independent bulk invoice

### PR DESCRIPTION
**Pull request by erikatras**

### CODE REVIEW

File: api/Controllers/ReportsController.cs [hunk 1]
```diff
@@ -308,54 +286,26 @@ private string GenerateSchoolYear(string scope)
...
-
-			var stamp = new Random(DateTime.Now.Millisecond).Next();
-
-			// create report
-			Report report;
-			using (var xlsxms = new MemoryStream())
-			using (var pdfms = new MemoryStream())
+			for (int i = 0; i < wb.Worksheets.Count; i++)
 			{
+<<<<<<< HEAD
```
Comment (willdmoore) [inline api/Controllers/ReportsController.cs]: I don't think you meant to leave these merge tags in...

File: api/Controllers/ReportsController.cs [hunk 2]
```diff
@@ -308,54 +286,26 @@ private string GenerateSchoolYear(string scope)
...
+					sheet.PageSetup.HeaderMargin = 0.0;
+					sheet.PageSetup.FooterMargin = 0.0;
+					sheet.PageSetup.TopMargin = 0.0;
+					sheet.PageSetup.TopMargin = 0.0;
+					sheet.PageSetup.LeftMargin = 0.0;
+					sheet.PageSetup.RightMargin = 0.0;
+					sheet.PageSetup.HeaderMargin = 0.0;
+					sheet.PageSetup.FooterMargin = 0.0;
+				}
+=======
```
Comment (willdmoore) [inline api/Controllers/ReportsController.cs]: See above.

File: api/Controllers/ReportsController.cs [hunk 3]
```diff
@@ -373,44 +323,36 @@ private Report CreateInvoice(DateTime time, Template invoiceTemplate, CreateRepo
 					Xlsx = xlsxms.ToArray(),
 					Pdf = pdfms.ToArray()
 				};
+>>>>>>> development
```
Comment (willdmoore) [inline api/Controllers/ReportsController.cs]: Etc.

File: api/Controllers/ReportsController.cs [hunk 4]
```diff
@@ -373,44 +323,36 @@ private Report CreateInvoice(DateTime time, Template invoiceTemplate, CreateRepo
...
-			for (int i = 0; i < invoices.Count; i++)
+
+			var districts = invoice.Districts.ToList();
+			for (int i = 0; i < districts.Count; i++)
 			{
-				var invoice = invoices[i];
+				var district = districts[i];
 
 				if (i == 0)
 				{
```
Comment (willdmoore) [inline api/Controllers/ReportsController.cs:338]: Is this supposed to be empty?  Looks like an artifact prior to the changes here.

File: api/Controllers/ReportsController.cs [hunk 5]
```diff
@@ -419,21 +361,23 @@ private Report CreateBulkInvoice(DateTime time, Template invoiceTemplate, Create
...
-			{
+			if (districts[0].Students.Count() == 0)
 				wb.Worksheets.RemoveAt(1);
-			}
-			// generate xlsx
 
 			foreach (var sheet in wb.Worksheets)
 				sheet.PageSetup.SetFooter(1, "&P");
 
+<<<<<<< HEAD
```
Comment (willdmoore) [inline api/Controllers/ReportsController.cs]: Merge tag

File: api/Controllers/ReportsController.cs [hunk 6]
```diff
@@ -419,21 +361,23 @@ private Report CreateBulkInvoice(DateTime time, Template invoiceTemplate, Create
...
+			// generate xlsx
+			var json = JsonConvert.SerializeObject(invoice);
+=======
```
Comment (willdmoore) [inline api/Controllers/ReportsController.cs]: Merge tag.

File: api/Controllers/ReportsController.cs [hunk 7]
```diff
@@ -459,6 +403,7 @@ private Report CreateBulkInvoice(DateTime time, Template invoiceTemplate, Create
 			};
 
 			var json = JsonConvert.SerializeObject(data);
+>>>>>>> development
```
Comment (willdmoore) [inline api/Controllers/ReportsController.cs]: Merge tag.

File: api/Controllers/ReportsController.cs [hunk 8]
```diff
@@ -508,55 +453,117 @@ private class MissingTemplateException : Exception
...
-
-			var name = create.Name + "_ACTIVITY";
-			var wb = BuildActivityWorkbook(name, new[] { sourceReport });
-
-			Report report;
-			using (var xlsxStream = new MemoryStream())
-			using (var pdfStream = new MemoryStream())
+			var report = CreateBulkInvoice(time, template, new CreateReport
 			{
+<<<<<<< HEAD
```
Comment (willdmoore) [inline api/Controllers/ReportsController.cs]: Merge tag.

File: api/Controllers/ReportsController.cs [hunk 9]
```diff
@@ -508,55 +453,117 @@ private class MissingTemplateException : Exception
...
+				ReportType = create.ReportType,
+				Name = create.Name,
+				SchoolYear = create.SchoolYear,
+				TemplateId = create.TemplateId,
+=======
 				wb.CalculateFormula();
 				wb.Save(xlsxStream, SaveFormat.Xlsx);
```
Comment (willdmoore) [inline api/Controllers/ReportsController.cs]: Merge tag.

File: api/Controllers/ReportsController.cs [hunk 10]
```diff
@@ -508,55 +453,117 @@ private class MissingTemplateException : Exception
...
 				wb.Save(pdfStream, SaveFormat.Pdf);
+>>>>>>> development
```
Comment (willdmoore) [inline api/Controllers/ReportsController.cs]: Merge tag.

**Comment by erikatras** [comment]
@willdmoore Yeah, I noticed the merge tags too late. :D

File: api/Controllers/ReportsController.cs (hunk 4 above)
Comment (erikatras) [reply api/Controllers/ReportsController.cs:338]: I'm not sure what this comment is in reference to. What's empty?

Comment (willdmoore) [reply api/Controllers/ReportsController.cs:338]: My head, apparently.  Ignore, please.

Comment (erikatras) [reply api/Controllers/ReportsController.cs:338]: 👍

**Pull request merged**